### PR TITLE
While debugging a Lambda function locally, make sure stopping the deb…

### PR DESCRIPTION
**Pull request by hunterwerlla**
…ugger will always stop the underlying SAM cli process 

See large comment on the reasoning of the change
## Related Issue(s)

#2564

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/steps/AttachDebugger.kt [hunk 1]
```diff
@@ -73,6 +98,7 @@ class AttachDebugger(
     }
 
     private companion object {
+        const val endText = "END RequestId: "
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/steps/AttachDebugger.kt:101]: do we have a place noting all of the magic SAM strings we're relying on?

**Review by rli — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/steps/AttachDebugger.kt (hunk 1 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/steps/AttachDebugger.kt:101]: Currently they are where they are used, but maybe we should move them into SamExecutable, thoughts?

**Comment by abrooksv** [comment]
Is it possible to write a test? For example, tweak an existing test so that when we hit a breakpoint, instead of resuming, we kill off the debugger session instead.

**Comment by hunterwerlla** [comment]
> Is it possible to write a test? For example, tweak an existing test so that when we hit a breakpoint, instead of resuming, we kill off the debugger session instead.

Added an integration test, still need to investigate the aparent dotnet image issue

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/steps/AttachDebugger.kt (hunk 1 above)
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/steps/AttachDebugger.kt:101]: yeah i like that idea

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/RunConfigTestUtils.kt [hunk 2]
```diff
@@ -89,6 +89,30 @@ fun executeRunConfigurationAndWait(runConfiguration: RunConfiguration, executorI
...
+            override fun processStarted(debugProcess: XDebugProcess) {
+                println("Debugger attached: $debugProcess")
+
+                debugProcess.session.addSessionListener(
+                    object : XDebugSessionListener {
+                        override fun sessionPaused() {
+                            runInEdt {
+                                val suspendContext = debugProcess.session.suspendContext
+                                println("Stopping: $suspendContext")
+                                callback()
```
Comment (abrooksv) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/RunConfigTestUtils.kt:105]: We dont seem to use the callback, do we need it?

**Review by abrooksv — APPROVED** [review]

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/RunConfigTestUtils.kt (hunk 2 above)
Comment (hunterwerlla) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/RunConfigTestUtils.kt]: Correct, removed

**Pull request merged**